apps/addons/models.py

```diff
@@ -508,12 +512,15 @@ def initialize_addon_from_upload(cls, data, is_listed=True):
         if not locale_is_set:
             addon.default_locale = to_language(trans_real.get_language())
 
+        addon.save()
+        if old_guid_addon:
+            old_guid_addon.update(guid='guid-reused-by-pk%s' % addon.pk)
```

Comment:
Could we have a `-` between the `pk` and the addon private key?
`'guid-reused-by-pk-%s'`

Also, we usually try to use `format` for string interpolation, so that would be:
`'guid-reused-by-pk-{}'.format(addon.pk)`

I have some concerns about that, for addons with long GUIDs: there's a limit of 255 chars, we need to make sure that we don't end up with something too long. I'd recommend truncating the resulting guid just to make sure. And maybe using a prefix a bit shorter? Something like `reused-by-`?